core/src/main/java/com/linecorp/armeria/common/ContentDisposition.java

```diff
@@ -0,0 +1,536 @@
+     * and {@code filename}.
+     *
+     * @param type the disposition type like for example {@code inline}, {@code attachment},
+     *             or {@code form-data}
+     * @param name the name parameter
+     * @param filename the filename parameter that will be be formatted as quoted-string,
+     *                 as defined in RFC 2616, section 2.2, and any quote characters within
+     *                 the filename value will be escaped with a backslash,
+     *                 e.g. {@code "foo\"bar.txt"} becomes {@code "foo\\\"bar.txt"}
+     */
+    public static ContentDisposition of(String type, String name, String filename) {
+        return builder(type).name(name).filename(filename).build();
+    }
+
+    /**
+     * Returns an empty {@link ContentDisposition}.
+     */
+    public static ContentDisposition of() {
+        return EMPTY;
+    }
+
+    /**
+     * Parses a {@code "content-disposition"} header value as defined in RFC 2183.
+     *
+     * <p>Note that only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+     *
+     * @param contentDisposition the {@code "content-disposition"} header value
+     * @return the parsed content disposition
+     * @see #asHeaderValue()
+     */
+    public static ContentDisposition parse(String contentDisposition) throws UnsupportedEncodingException {
+        requireNonNull(contentDisposition, "contentDisposition");
+        final List<String> parts = tokenize(contentDisposition);
+        final String type = parts.get(0);
+        String name = null;
+        String filename = null;
+        Charset charset = null;
+        Long size = null;
+        ZonedDateTime creationDate = null;
+        ZonedDateTime modificationDate = null;
+        ZonedDateTime readDate = null;
+        for (int i = 1; i < parts.size(); i++) {
+            final String part = parts.get(i);
+            final int eqIndex = part.indexOf('=');
+            if (eqIndex != -1) {
+                final String attribute = part.substring(0, eqIndex);
+                final String value;
+                if (part.startsWith("\"", eqIndex + 1) && part.endsWith("\"")) {
+                    value = part.substring(eqIndex + 2, part.length() - 1);
+                } else {
+                    value = part.substring(eqIndex + 1);
+                }
+
+                if ("name".equals(attribute)) {
+                    name = value;
+                } else if ("filename*".equals(attribute)) {
+                    final int idx1 = value.indexOf('\'');
+                    final int idx2 = value.indexOf('\'', idx1 + 1);
+                    if (idx1 != -1 && idx2 != -1) {
+                        charset = Charset.forName(value.substring(0, idx1).trim());
+                        checkArgument(UTF_8.equals(charset) || ISO_8859_1.equals(charset),
+                                      "Charset: %s (expected: UTF-8 or ISO-8859-1)", charset);
```

Comment:
Should we raise an exception or fall back to ISO-8859-1?

Reply:
There might be no strong restriction in RFC 6266. https://tools.ietf.org/html/rfc6266#section-4.3
I think we can fall back to `ISO-8859-1`.

Reply:
Sorry for this part. I have just read RFC5987(https://tools.ietf.org/html/rfc5987#section-3.2.1). It says:
> Producers MUST use either the "UTF-8" ([RFC3629]) or the "ISO-8859-1"
   ([ISO-8859-1]) character set.  Extension character sets (mime-
   charset) are reserved for future use.


Is it better to limit the supported charsets to "UTF-8" and "ISO-8859-1" only?

Reply:
Yes, please. :-)